File: seungmi/week10/n으로표현.py
def solution(N, number):
    
    if N==number:
        return 1
    
    s=[set() for x in range(8)]
    
    for i,s_set in enumerate(s, start=1):
        s_set.add(int(str(N)*i))
        
    for i in range(1, 8):
        for j in range(i):
            for op1 in s[j]:
                for op2 in s[i-j-1]:
                    s[i].add(op1 + op2)
                    s[i].add(op1 - op2)
                    s[i].add(op1 * op2)
                    if op2 != 0:
                        s[i].add(op1 // op2)

        if  number in s[i]:
            answer = i + 1
            break

    else:
        answer = -1

    return answer

File: sujeong/week10/N으로 표현.swift
import Foundation

func solution(_ N:Int, _ number:Int) -> Int {
    
    var setOf:[Set<Int>] = Array(repeating: Set<Int>(), count:9)
    
    for i in 1...8 {
        setOf[i].insert(getConcat(N, i))
        
        if setOf[i].contains(number) { return i }
        if i == 1 { continue }
        
        for j in 1...(i-1) {
            for op1 in setOf[j] {
                for op2 in setOf[i-j] {
                    setOf[i].insert(op1 + op2)
                    setOf[i].insert(op1 - op2)
                    setOf[i].insert(op1 * op2)
                    if op2 != 0 {
                        setOf[i].insert(op1/op2)
                    }
                }
            }
        }
        if setOf[i].contains(number) { return i }
    }
    return -1
}

func getConcat(_ N:Int, _ i:Int) -> Int {
    var result:Int = 0
    for n in 0..<i {
        result = result*10 + N
    }
    return result
}
